recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,171 @@
+                self.options["cpython"].shared = True
+        else:
+            self.options.rm_safe("python_version")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("tomlplusplus/3.2.0")
+        if self.options.with_python:
+            self.requires("openssl/1.1.1s")
+            self.requires(f"cpython/{self.options.python_version}")
+            self.requires("pybind11/2.10.1")
+
+    def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support MacOS at this time"
+            )
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.1")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["OPENASSETIO_ENABLE_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+        tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = self.settings.get_safe("compiler.libcxx") == "libstdc++11"
+
+        tc.variables["OPENASSETIO_ENABLE_PYTHON"] = self.options.with_python
+        if self.options.with_python:
+            tc.variables["Python_EXECUTABLE"] = self._python_exe
+            if is_msvc(self):
+                tc.variables["Python_LIBRARY"] = self._python_windows_lib
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    @property
+    def _python_exe(self):
+        return pathlib.Path(self.deps_user_info["cpython"].python).as_posix()
+
+    @property
+    def _python_windows_lib(self):
+        pth = pathlib.Path(
+            self.dependencies["cpython"].package_folder,
+            self.dependencies["cpython"].cpp_info.components["embed"].libdirs[0],
+            self.dependencies["cpython"].cpp_info.components["embed"].libs[0])
+        pth = pth.with_suffix(".lib")
+        return pth.as_posix()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package_id(self):
+        if self.options.with_python:
+            self.info.requires["cpython"].minor_mode()
+
```

Comment:
```suggestion
```
Forcing the package ID mode in CCI is not allowed because it will cause missing packages. Each package ID mode will result in a different hash, which is not be available in the server, so when having a change and needing to build again OpenAssetIO, cpython will not be available as pre-built.